src/DocumentFormat.OpenXml.Framework/OpenXmlWriter.cs

```diff
@@ -123,13 +126,86 @@ protected OpenXmlWriter()
         /// <param name="text">The text to write. </param>
         public abstract void WriteString(string text);
 
-        // public abstract void WriteRaw(string data);
-
         /// <summary>
         /// Close the writer.
         /// </summary>
         public abstract void Close();
 
+#if FEATURE_ASYNC_SAX_XML
+        /// <summary>
+        /// Asynchronously writes the XML declaration with the version "1.0".
+        /// </summary>
+        public virtual Task WriteStartDocumentAsync()
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Asynchronously writes the XML declaration with the version "1.0" and the standalone attribute.
+        /// </summary>
+        /// <param name="standalone">If true, it writes "standalone=yes"; if false, it writes "standalone=no". </param>
+        public virtual Task WriteStartDocumentAsync(bool standalone)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Asynchronously writes out a start tag of the element and all the attributes of the element.
+        /// </summary>
+        /// <param name="elementObject">The OpenXmlElement object to be written.</param>
+        public virtual Task WriteStartElementAsync(OpenXmlElement elementObject)
+        {
+            throw new NotImplementedException();
```

Comment:
instead of throwing, it may be helpful to implement against sync method. They're new methods, but any implementations of OpenXmlWriter that doesn't override them will throw, so I think we should default to an implementation that won't throw